include/felspar/parse/extract.detail.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+
+#include <span>
+
+
+namespace felspar::parse::binary::detail {
+
+
+    template<typename T>
+    inline T native_extract(std::span<std::byte const, sizeof(T)> const s) noexcept {
+        T into;
+        std::copy(s.begin(), s.end(), reinterpret_cast<std::byte *>(&into));
```

Comment:
Is that _legal?_  It doesn't need any kind of memory barrier to tell the compiler that there's deliberate aliasing of incompatible types?  Would `std::as_writable_bytes` help with that?

Also, how portable do you need the communication to be?  C++ now guarantees 2's complement for integers... but I don't think it guarantees much about floating-point formats.

Reply:
There's likely some weird platforms where floats aren't IEEE, but I don't think that'll be the case for any that I care about. The stand out is Windows where apparently `long double` isn't 128 bits. That may make that type non-portable, although that's not really anything to do with this library.

https://en.cppreference.com/w/cpp/language/types